src/components/data-exploration/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -150,13 +159,21 @@ const DiffExprCompute = (props) => {
           numCellsPerSampleInCellSet[sampleIdx] += 1;
         });
 
-      return numCellsPerSampleInCellSet.find(numCells => numCells > MIN_NUM_CELLS_IN_GROUP)
+      return numCellsPerSampleInCellSet.filter(numCells => numCells >= MIN_NUM_CELLS_IN_GROUP).length;
     }
 
-    const cellSetHasSampleWithEnoughCells = hasSampleWithEnoughCells(filteredCellSetCellIds)
-    const compareWithHasSampleWithEnoughCells = hasSampleWithEnoughCells(filteredCompareWithCellIds)
+    const numCellSetSampleWithEnoughCells = numSampleWithEnoughCells(filteredCellSetCellIds)
```

Comment:
`canRunDiffExpr` seems pretty long and complex, do you think it could be extracted to a function an imported here? And if there's also other functions like this maybe they should be changed too.

The jsx file could just have the logic more directly related to what to render and call functions for the processing, etc.

Reply:
This logic is only used in this file. It is also tightly coupled with the variables and state of the function, so I think there's no need to extract it out.

Some of the cell set operations inside the function is repeated a lot throughout the code base though. For these operations, I agree we can factor it out into separate functions. However, I think it is a general refactor better left for another ticket / hackathon.

Reply:
I agree, we don't need to do a general refactor on the cell set operations here.

My reason for separating responsibilities was not mainly about reusability, more about readability and being able to have a clearer separation, an example of something similar could be `processUpload` or `calculateVolcanoDataPoints`, they are all about doing more involved calculations and they are extracted into functions even when we could easily just have them i one place in the `jsx` that uses them.